Sources/YUL/AST.swift

```diff
@@ -0,0 +1,342 @@
+//
```

Comment:
Can we split this across multiple files?

Reply:
AST is also too general of a name. Each enum/struct could potentially live in its own file or we group them by parser rule (statement, expression, etc) and _name the file accordingly_.